File: ForegroundService/app/src/main/java/com/example/foregroundservice/MainActivity.java
package com.example.foregroundservice;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
public class MainActivity extends AppCompatActivity {

    private EditText edtDataInt;
    private Button btnStartService;
    private Button btnStopService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtDataInt = findViewById(R.id.edt_data_intent);
        btnStartService = findViewById(R.id.btn_start_service);
        btnStopService = findViewById(R.id.btn_stop_service);
        
        btnStartService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickStartService();
            }
        });

        btnStopService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickStopService();
            }
        });

    }

    private void clickStopService() {
        Intent intent = new Intent(this, MyService.class);
        //        khoi chay bang cau lenh
// Khi công việc hoàn thành bạn nên stop bằng cách gọi stopService() từ một thành phần khác, hoặc cho chính Service gọi stopSelf().
//     hàm nào được gọi trong main khi nào với cái loại service nào(loại service là cái hàm ovirride bên service đó)
        stopService(intent);
    }

    private void clickStartService() {
//        Song song = new Song("city", "kk", R.drawable.ic_launcher_foreground, R.raw.sommai);

        Intent intent = new Intent(this, MyService.class);
//        Bundle bundle = new Bundle();
//        bundle.putSerializable("object_song", song);
//        intent.putExtras(bundle) ;
// truyền dữ liệu lên service thì xử lý dữ liệu ở hàm onStartComand bên service
                intent.putExtra("data_intent", edtDataInt.getText().toString().trim());
        //        khoi chay bang cau lenh
        startService(intent);
    }

}

File: ForegroundService_KOTLIN/app/src/main/java/com/example/foregroundservice_kotlin/MainActivity.kt
package com.example.foregroundservice_kotlin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.example.foregroundservice_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding  = ActivityMainBinding.inflate(layoutInflater)

//        binding.setOnClickListener(View.OnClickListener { clickStartService() })
//
//        btnStopService.setOnClickListener(View.OnClickListener { clickStopService() })
        binding.btnStopService.setOnClickListener {
            clickStopService()
        }
        binding.btnStartService.setOnClickListener {
            clickStartService()
        }
        setContentView(binding.root)
    }

    private fun clickStopService() {
        val intent = Intent(this, MyService::class.java)
        //        khoi chay bang cau lenh
// Khi công việc hoàn thành bạn nên stop bằng cách gọi stopService() từ một thành phần khác, hoặc cho chính Service gọi stopSelf().
//     hàm nào được gọi trong main khi nào với cái loại service nào(loại service là cái hàm ovirride bên service đó)
        stopService(intent)
    }

    private fun clickStartService() {
//        Song song = new Song("city", "kk", R.drawable.ic_launcher_foreground, R.raw.sommai);

        val intent = Intent(this, MyService::class.java)
        //        Bundle bundle = new Bundle();
//        bundle.putSerializable("object_song", song);
//        intent.putExtras(bundle) ;
// truyền dữ liệu lên service thì xử lý dữ liệu ở hàm onStartComand bên service
        intent.putExtra("data_intent", binding.edtDataIntent.text.toString().trim())
        //        khoi chay bang cau lenh
//        Gọi startService thì nó chạy vào startCommand bên Myservice
//        Tương tự mấy hàm gọi khác
        startService(intent)
    }

}